fe/fe-core/src/main/java/com/starrocks/privilege/external/ConfigurableExternalAccessController.java

```
@@ -0,0 +1,62 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.privilege.external;
+
+import com.starrocks.common.Config;
+import com.starrocks.privilege.AccessDeniedException;
+import com.starrocks.privilege.ExternalAccessController;
+import com.starrocks.privilege.PrivilegeType;
+import com.starrocks.privilege.external.ExternalAccessResourceImpl;
+import com.starrocks.privilege.external.ExternalStarRocksAccessRequest;
+import com.starrocks.sql.ast.UserIdentity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ConfigurableExternalAccessController extends ExternalAccessController implements AccessTypeConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigurableExternalAccessController.class);
+    protected ExternalAuthorizer authorizer;
+
+    public ConfigurableExternalAccessController() {
+        super(false);
+        authorizer = initExternalAuthorizer();
+    }
+
+    private ExternalAuthorizer initExternalAuthorizer() {
+        String externalAuthorizationClassName = Config.external_authorization_class_name;
+        try {
+            ExternalAuthorizer authorizer = Class.forName(externalAuthorizationClassName)
+                            .asSubclass(ExternalAuthorizer.class).newInstance();
+            authorizer.init();
+            return authorizer;
+        } catch (Exception e) {
+            LOG.error("Failed to create external authorizer", e);
```

I see that we fail later in the hasPermission function, but my instinct is that we should fail loudly at construction instead of during use. I would probably say that we should throw some sort of uncaught exception if we cannot create the authorizer. We could also fail in another way, but we don't want to succeed in constructing the ConfigurableExternalAccessController if the authorizer is null, as we would right now.

---

Understood. Makes sense.

---

Updated to throw a runtime exception on failure to initialize.